src/main/java/com/minecolonies/coremod/colony/requestsystem/locations/StaticLocation.java

```diff
@@ -0,0 +1,154 @@
+     */
+    public static class Factory implements ILocationFactory<BlockPos, StaticLocation>
+    {
+
+        ////// --------------------------- NBTConstants --------------------------- \\\\\\
+        private static final String NBT_POS = "Pos";
+        private static final String NBT_DIM = "Dim";
+        ////// --------------------------- NBTConstants --------------------------- \\\\\\
+
+        @NotNull
+        @Override
+        public TypeToken<StaticLocation> getFactoryOutputType()
+        {
+            return new TypeToken<StaticLocation>() {};
+        }
+
+        @NotNull
+        @Override
+        public TypeToken<BlockPos> getFactoryInputType()
+        {
+            return new TypeToken<BlockPos>() {};
+        }
+
+        /**
+         * Method to serialize a given constructable.
+         *
+         * @param controller The controller that can be used to serialize complicated types.
+         * @param request    The request to serialize.
+         * @return The serialized data of the given requets.
+         */
+        @NotNull
+        @Override
+        public NBTTagCompound serialize(@NotNull IFactoryController controller, @NotNull StaticLocation request)
+        {
+            NBTTagCompound compound = new NBTTagCompound();
+            compound.setLong(NBT_POS, request.getInDimensionLocation().toLong());
+            compound.setInteger(NBT_DIM, request.getDimension());
+            return compound;
+        }
+
+        /**
+         * Method to deserialize a given constructable.
+         *
+         * @param controller The controller that can be used to deserialize complicated types.
+         * @param nbt        The data of the request that should be deserialized.
+         * @return The request that corresponds with the given data in the nbt
+         */
+        @NotNull
+        @Override
+        public StaticLocation deserialize(@NotNull IFactoryController controller, @NotNull NBTTagCompound nbt)
+        {
+            BlockPos pos = BlockPos.fromLong(nbt.getLong(NBT_POS));
+            Integer dim = nbt.getInteger(NBT_DIM);
+            return new StaticLocation(pos, dim);
+        }
+
+        @NotNull
+        @Override
+        public StaticLocation getNewInstance(@NotNull final BlockPos blockPos, @NotNull final Object... context) throws IllegalArgumentException
+        {
+            if (context.length != 1)
+            {
+                throw new IllegalArgumentException("Unsupported context - Not the correct amount available. Needed is 1!");
+            }
+
+            if (!(context[0] instanceof Integer))
+            {
+                throw new IllegalArgumentException("Unsupported context - First context object is not a Integer. Provide an Integer as Dimension.");
+            }
+
+            return new StaticLocation(blockPos, (Integer) context[0]);
+        }
+
+        /**
+         * Method to get a new instance of a location given the input.
+         * <p>
+         * <p>
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)